### PR TITLE
Added new test plans for VNET to VNET testing

### PR DESCRIPTION
**Pull request by murthyvijay**
* SONiC test plan template is placed here and can be used as a reference for future test plans
* All future DASH test plans may follow the same template; ensures uniformity
* Included images to denote testbed topology and ixia-c based test setup deployment
* Basic feature functionality of Virtual Networks (VNET) is tested while traffic generator emulate various traffic patterns
* Covers test cases written for:
  * Traffic originating in one VNET and terminating on another VNET and observing functionality of the DUT
  * Communication across two virtual machines in different VNETs
  * Virtual Machines with IPv4/IPv6 addressing
  * ACL features like allow and deny

### CODE REVIEW

**Comment by murthyvijay** [comment]
Keysight

File: test/docs/testplans/vnet_to_vnet.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+| UDP IPv6 outbound direction | Verify traffic originating from one VNet simulated terminates on another simulated VNet. <br><br> Verify counters in DUT in the inbound direction are accurately incremented. <br><br> Verify latency measurements <br> Verify packet loss is 0 (Tx frames - Rx frames = 0) <br><br> Verify configured throughput rate is achieved in traffic statistics  |
+| UDP IPv6 bidirectional  | Verify traffic originating from one VNet simulated terminates on another simulated VNet. <br><br> Verify counters in DUT in the inbound direction are accurately incremented. <br><br> Verify latency measurements <br> Verify packet loss is 0 (Tx frames - Rx frames = 0) <br><br> Verify configured throughput rate is achieved in traffic statistics  |                                                 
+|  ACL inbound | Verify data traffic is allowed and unwanted traffic based on ACL is dropped. <br><br>Verify ACL inbound counters on DUT are incremented
```
Comment (chrispsommers) [inline test/docs/testplans/vnet_to_vnet.md:56]: Please specify whether IPv4 or IPv6 or both; if both, suggest you have table row for each permutation [IPv4, IPv6] x [Inbound, Outbound] i.e. 4 test cases.

File: test/docs/testplans/vnet_to_vnet.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+- Traffic generator will be used to configure traffic streams that originate and/or terminate on traffic generator ports
+- Data traffic will be sent from VNET to VNET to verify performance properties like: CPS, flow, PPS, and rule scale.  These performance properties will be verified while  features like: VM to VM communication in VNET, route support, LPM support, ACL support are enabled.
+- Depending on the test cases, a combination of UDP traffic for IPv4, IPv6 addresses will be generated
+
+### Test Cases
+[TODO: Update Table]
+
+| Test Description                                             | Expected Result               |
+| :----------------------------------------------------------- | :------------------------ |
+| UDP IPv4 inbound  | Verify traffic originating from one VNet simulated terminates on another simulated VNet. <br><br> Verify counters in DUT in the inbound direction are accurately incremented. <br><br> Verify latency measurements <br> Verify packet loss is 0 (Tx frames - Rx frames = 0) <br><br> Verify configured throughput rate is achieved in traffic statistics  |
```
Comment (chrispsommers) [inline test/docs/testplans/vnet_to_vnet.md:50]: Please specify what assumptions, if any, which you make about counters on the DUT. No counters exist to date. We should feed expectations into the working groups so we can track items. It is also OK to say "once counters are defined, the test-cases will be refined to incorporate them."

**Review by chrispsommers — CHANGES_REQUESTED** [review]
Minor clarifications requested, thanks.

File: test/docs/testplans/test_plan_template.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+#Test name
```
Comment (lguohan) [inline test/docs/testplans/test_plan_template.md:1]: please separate this template into a different pr

File: test/docs/testplans/vnet_to_vnet.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+### Testbed
+
+The generic testbed setup may consist of two OTG connected to two ports of a switch (hardware-based or BMv2 software switch). 
+In this specifc testbed setup we will use [Ixia-c](https://github.com/open-traffic-generator/ixia-c) which is a modern, powerful, and API-driven open traffic generator model-based traffic tool that will generate traffic to observe the behavior of the device under test (DUT).
+
+### Testbed Topology 
+
+The tesbed is setup to emulate a VNET to VNET scenario and to test the mechanisms of VM to VM communication in VNET as shown below.
+
+![VM-VM-in-VNET](../../images/vm-to-vm-communication-vnet.svg)
```
Comment (lguohan) [inline test/docs/testplans/vnet_to_vnet.md:27]: this diagram is a little bit misleading. ideally, the packet flow should be 

vm1 <--> dash1 <---> dash2 <--> vm2

File: test/docs/testplans/test_plan_template.md (hunk 3 above)
Comment (murthyvijay) [reply test/docs/testplans/test_plan_template.md:1]: This template was borrowed from [SONiC test plan template](https://github.com/sonic-net/SONiC/blob/master/doc/SONiC%20Test%20Plan%20Template.md) and placed here for reference. I can leave it out if not deemed necessary.

File: test/docs/testplans/vnet_to_vnet.md (hunk 1 above)
Comment (murthyvijay) [reply test/docs/testplans/vnet_to_vnet.md:56]: Each IPv4 and IPv6 inbound/outbound is mentioned separately. As we add more testcases to the table all the permutation options will be mentioned.

File: test/docs/testplans/test_plan_template.md (hunk 3 above)
Comment (chrispsommers) [reply test/docs/testplans/test_plan_template.md:1]: Just make a link to this external document instead of duplicating it.

Comment (murthyvijay) [reply test/docs/testplans/test_plan_template.md]: fixed as per recommendation to link it to external document.

File: test/docs/testplans/vnet_to_vnet.md (hunk 4 above)
Comment (murthyvijay) [reply test/docs/testplans/vnet_to_vnet.md]: Linked to the original vnet to vnet service document under DASH and removed the SVG file

File: test/docs/testplans/vnet_to_vnet.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+### Testbed Topology 
+
+The testbed is setup to emulate a VNET to VNET scenario and to test the mechanisms of VM to VM communication in VNET as outlined [here](https://github.com/Azure/DASH/blob/main/documentation/vnet2vnet-service/vnet-to-vnet-service.md).
+
+
+
+To emulate each VM in a VNET an Ixia-c port will be utilized. The BMv2 will be the DUT (This can later be replaced by a hardware-based appliance that connects to physical ports of a traffic generator)
+
+![ixia-c.drawio](../../images/ixia-c.drawio.svg)
```
Comment (chrispsommers) [inline test/docs/testplans/vnet_to_vnet.md:31]: Why not use https://github.com/Azure/DASH/blob/main/test/third-party/traffic_gen/deployment/ixia-c.drawio.svg? It's actually more accurate.

**Review by chrispsommers — CHANGES_REQUESTED** [review]
Please reuse existing diagram unless there is an issue.

File: test/docs/testplans/vnet_to_vnet.md (hunk 5 above)
Comment (murthyvijay) [reply test/docs/testplans/vnet_to_vnet.md]: Updated the image as per above recommendation

**Review by chrispsommers — APPROVED** [review]

**Comment by KrisNey-MSFT** [comment]
@xumia can you help us with the new Easy CLA check that was added (and PRd now not passing) please?
